components/lib/crash/src/test/java/mozilla/components/lib/crash/service/GleanCrashReporterServiceTest.kt

```diff
@@ -0,0 +1,112 @@
+    private val context: Context
+        get() = ApplicationProvider.getApplicationContext()
+
+    private fun setupGleanCrashReporterServiceTest() {
+        WorkManagerTestInitHelper.initializeTestWorkManager(
+            ApplicationProvider.getApplicationContext())
+
+        Glean.enableTestingMode()
+        Glean.initialize(context)
+    }
+
+    @Test
+    fun `GleanCrashReporterService records native code crashes`() {
+        run {
+            val service = spy(GleanCrashReporterService(context))
+
+            assertFalse("No previous persisted crashes must exist", service.file.exists())
+
+            val crash = Crash.NativeCodeCrash("", true, "", false)
+            service.report(crash)
+
+            verify(service).report(crash)
+
+            assertTrue("Persistence file must exist", service.file.exists())
+            val lines = service.file.readLines()
+            assertEquals("Must be uncaught exception",
+                GleanCrashReporterService.NATIVE_CODE_CRASH_KEY, lines.first())
+        }
+
+        // Initialize a fresh GleanCrashReporterService and ensure metrics are recorded in Glean
+        run {
+            setupGleanCrashReporterServiceTest()
+            GleanCrashReporterService(context)
+
+            assertTrue("Glean must record a value",
+                CrashMetrics.crashCount[GleanCrashReporterService.UNCAUGHT_EXCEPTION_KEY].testHasValue())
+            assertEquals("Glean must record correct value",
+                1,
+                CrashMetrics.crashCount[GleanCrashReporterService.UNCAUGHT_EXCEPTION_KEY].testGetValue())
+        }
+    }
+
+    @Test
+    fun `GleanCrashReporterService records uncaught exceptions`() {
+        run {
+            val service = spy(GleanCrashReporterService(context))
+
+            assertFalse("No previous persisted crashes must exist", service.file.exists())
+
+            val crash = Crash.UncaughtExceptionCrash(RuntimeException("Test"))
+            service.report(crash)
+
+            verify(service).report(crash)
+
+            assertTrue("Persistence file must exist", service.file.exists())
+            val lines = service.file.readLines()
+            assertEquals("Must be uncaught exception",
+                GleanCrashReporterService.UNCAUGHT_EXCEPTION_KEY, lines.first())
+        }
+
+        // Initialize a fresh GleanCrashReporterService and ensure metrics are recorded in Glean
+        run {
+            setupGleanCrashReporterServiceTest()
+            GleanCrashReporterService(context)
+
+            assertTrue("Glean must record a value",
+                CrashMetrics.crashCount[GleanCrashReporterService.UNCAUGHT_EXCEPTION_KEY].testHasValue())
+            assertEquals("Glean must record correct value",
+                1,
+                CrashMetrics.crashCount[GleanCrashReporterService.UNCAUGHT_EXCEPTION_KEY].testGetValue())
+        }
+    }
+
+    @Test
+    fun `GleanCrashReporterService is a no-op if Glean is not initialized`() {
```

Comment:
Can we add a test that records both multiple native crashes and uncaught exceptions?